services/rollout-service/pkg/service/broadcast_test.go

```diff
@@ -437,3 +483,140 @@ func TestBroadcastDoesntGetStuck(t *testing.T) {
+					SyncStatusCode:   v1alpha1.SyncStatusCodeSynced,
+					HealthStatusCode: health.HealthStatusHealthy,
+				},
+				{
+					Application:      "foo",
+					Environment:      "prd",
+					Version:          &versions.VersionInfo{Version: 1},
+					SyncStatusCode:   v1alpha1.SyncStatusCodeSynced,
+					HealthStatusCode: health.HealthStatusHealthy,
+				},
+			},
+			KuberpultEvents: []versions.KuberpultEvent{
+				{
+					Application:      "foo",
+					Environment:      "dev",
+					Version:          &versions.VersionInfo{Version: 3},
+					EnvironmentGroup: "dev-group",
+				},
+				{
+					Application:      "foo",
+					Environment:      "prd",
+					Version:          &versions.VersionInfo{Version: 3},
+					EnvironmentGroup: "prd-group",
+				},
+			},
+			Request: &api.GetStatusRequest{
+				EnvironmentGroup: "dev-group",
+			},
+			ExpectedResponse: &api.GetStatusResponse{
+				Status: api.RolloutStatus_RolloutStatusPending,
+				Applications: []*api.GetStatusResponse_ApplicationStatus{
+					{
+						Environment:   "dev",
+						Application:   "foo",
+						RolloutStatus: api.RolloutStatus_RolloutStatusPending,
+					},
+				},
+			},
+		},
+		{
+			Name: "exludes succesful applications",
+			ArgoEvents: []ArgoEvent{
+				{
+					Application:      "foo",
+					Environment:      "dev",
+					Version:          &versions.VersionInfo{Version: 2},
+					SyncStatusCode:   v1alpha1.SyncStatusCodeSynced,
+					HealthStatusCode: health.HealthStatusHealthy,
+				},
+				{
+					Application:      "bar",
+					Environment:      "dev",
+					Version:          &versions.VersionInfo{Version: 1},
+					SyncStatusCode:   v1alpha1.SyncStatusCodeSynced,
+					HealthStatusCode: health.HealthStatusHealthy,
+				},
+			},
+			KuberpultEvents: []versions.KuberpultEvent{
+				{
+					Application:      "foo",
+					Environment:      "dev",
+					Version:          &versions.VersionInfo{Version: 3},
+					EnvironmentGroup: "dev-group",
+				},
+				{
+					Application:      "bar",
+					Environment:      "dev",
+					Version:          &versions.VersionInfo{Version: 1},
+					EnvironmentGroup: "dev-group",
+				},
+			},
+			Request: &api.GetStatusRequest{
+				EnvironmentGroup: "dev-group",
+			},
+			ExpectedResponse: &api.GetStatusResponse{
+				Status: api.RolloutStatus_RolloutStatusPending,
+				Applications: []*api.GetStatusResponse_ApplicationStatus{
+					{
+						Environment:   "dev",
+						Application:   "foo",
+						RolloutStatus: api.RolloutStatus_RolloutStatusPending,
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			bc := New()
+			for _, s := range tc.ArgoEvents {
+				bc.ProcessArgoEvent(context.Background(), s)
+			}
+			for _, s := range tc.KuberpultEvents {
+				bc.ProcessKuberpultEvent(context.Background(), s)
+			}
+
+			ctx, cancel := context.WithCancel(context.Background())
+			resp, err := bc.GetStatus(ctx, tc.Request)
+			cancel()
```

Comment:
I'm not really familiar with this function here

Reply:
Hmm, right, It's also not really needed here. I'll remove it